letta/agent.py

```diff
@@ -596,6 +596,12 @@ def _get_ai_reply(
             self.functions if not allowed_tool_names else [func for func in self.functions if func["name"] in allowed_tool_names]
         )
 
+        # For the first message, force the initial tool if one is specified
+        force_tool_call = None
+        if first_message and self.tool_rules_solver.init_tool_rules:
+            assert len(self.tool_rules_solver.init_tool_rules) == 1, "Multiple initial tools not supported"
```

Comment:
maybe move this into the create agent function? and we should allow this (not throw an error) if the LLM config specifies OpenAI + one of these structured output models https://platform.openai.com/docs/guides/structured-outputs#supported-models

Reply:
Moved the check to a function at initialization `self.check_tool_rules()`